shapeshift/web
# fix: eth/fox LP approval

## Description

This PR:

- tackles the UI displaying an error after approval of ETH/FOX LP because of positional arguments `token0Amount` and `token1Amount` being inverted. Changed to a destructured object so we can avoid this in the future
- revamps the `hasEnoughBalanceForGas` flow at approval step to use an `useEffect` field, so we can set it to false in case of a failing estimate Tx, which would indicate a failed gas estimation 

## Notice




## Pull Request Type


## Issue (if applicable)



closes https://github.com/shapeshift/web/issues/4021

## Risk



Low, 

## Testing



- Make sure you have no / not enough allowance for ETH/FOX LP
- Go to deposit and be greeted with the allowance modal
- Ensure the approval goes through without any error toast, and allows you to continue to deposit step

Note, the gas estimation logic is currently wrong and in case you don't have enough ETH for approval gas / deposit, you will still be able to continue - fixed in https://github.com/shapeshift/web/pull/4024

### Engineering



- Refactor looks sane

### Operations



## Screenshots (if applicable)
